File: goofy-protocol-fis-be/src/main/java/com/masl/goofy_protocol_core/crypto/isolated/asymm/AsymmCryptoType.java
package com.masl.goofy_protocol_core.crypto.isolated.asymm;

import lombok.Getter;

@Getter
public enum AsymmCryptoType {
    // Using RSA/ECB/OAEPWithSHA-256AndMGF1Padding
    RSA_2048((short)0x0100),
    RSA_3072((short)0x0101),
    RSA_4096((short)0x0102),

    // Using P-256/384
    EC_P256((short)0x0200),
    EC_P384((short)0x0201),
    // Using Curve25519
    EC_C25519((short)0x0210),

    MLKEMDSA_512_44((short)0x0300),
    MLKEMDSA_768_65((short)0x0301),
    MLKEMDSA_1024_87((short)0x0302);

    private final short value;
    AsymmCryptoType(short value) {
        this.value = value;
    }

    public static AsymmCryptoType fromValue(short value) {
        for (AsymmCryptoType t : values())
            if (t.value == value)
                return t;
        throw new IllegalArgumentException("Unknown AsymmCryptoType value: " + value);
    }
}

File: goofy-protocol-fis-test-fe/libs/goofy-protocol-core/crypto/isolated/asymm/asymm-crypto-type.mjs
export const AsymmCryptoType = Object.freeze({
    RSA_2048: { name: "RSA_2048", value: 0x0100 },
    RSA_3072: { name: "RSA_3072", value: 0x0101 },
    RSA_4096: { name: "RSA_4096", value: 0x0102 },

    EC_P256: { name: "EC_P256", value: 0x0200 },
    EC_P384: { name: "EC_P384", value: 0x0201 },

    EC_C25519: { name: "EC_C25519", value: 0x0210 },

    MLKEMDSA_512_44: { name: "MLKEMDSA_512_44", value: 0x0300 },
    MLKEMDSA_768_65: { name: "MLKEMDSA_768_65", value: 0x0301 },
    MLKEMDSA_1024_87: { name: "MLKEMDSA_1024_87", value: 0x0302 },

    fromValue(value) {
        const keys = Object.keys(this).filter(k => k !== "fromValue");
        for (const k of keys) {
            const t = this[k];
            if (t.value === value) return t;
        }
        throw new Error(`Unknown AsymmCryptoType value: ${value}`);
    },
});